tuf/api/metadata.py

```diff
@@ -92,12 +93,17 @@ class also that has a 'from_dict' factory method. (Currently this is
         else:
             raise ValueError(f'unrecognized metadata type "{_type}"')
 
+        signatures = []
+        for signature in metadata['signatures']:
+            new_signature = Signature(signature['keyid'], signature['sig'])
```

Comment:
Two minor nits:
- I think the data type is a more meaningful distinguishing factor than the instantiation time.
  What do you think about `signature`/`signature_obj`,  or `signature_dict`/`signature`, or `signature_dict`/`signature_obj` (note: I am aware that a dict is also an object, but still). 
- Why not call `Signature.from_dict`? Then we don't have to know here what's going on inside a signature dictionary.

Reply:
Both of those remarks make sense. 
I totally forgot that I could use `Signature.from_dict()` and `signature_obj` sounds. like a reasonable name.